src/ia-anniversary-banner.ts

```diff
@@ -0,0 +1,194 @@
+import { html, css, LitElement, property, customElement } from 'lit-element';
+import { nothing } from 'lit-html';
+import CloseCircleIcon from '@internetarchive/icon-close-circle';
+
+import { IAFile, IAMD, TimelineMoment } from './interfaces';
+
+import { formatMoments } from './utils';
+import wayforwardSvg from './images/way-forward-svg';
+import wayforwardSvgMin from './images/way-forward-min-svg';
+
+import './marquee';
+
+@customElement('ia-anniversary-banner')
+export class IaAnniversaryBanner extends LitElement {
+  @property({ type: Object }) iaFiles: IAFile[] = [];
+
+  @property({ type: Object }) iaMD: IAMD = { directory: '', separator_dir: '' };
+
+  @property({ type: Object }) moments: { [key: string]: TimelineMoment } = {};
+
+  @property({ type: Array }) shuffledMoments: TimelineMoment[] = [];
+
+  @property({ type: String }) landingURL = 'https://archive.org';
+
+  @property({ type: String }) viewMode = 'open'; // open | closed
+
+  @property({ type: Number }) hideBannerDays = 7;
+
+  @property({ type: String }) directory = '';
+
+  updated(changed: any) {
+    if (changed.has('iaFiles') || changed.has('iaMD')) {
+      this.shuffledMoments = [];
+      const { directory } = this.iaMD;
+      this.directory = directory;
+    }
+    if (changed.has('directory')) {
+      this.shuffleMoments(this.directory);
+    }
+  }
+
+  shuffleMoments(directory: string) {
+    const moments: TimelineMoment[] = formatMoments(
+      this.iaFiles,
+      this.iaMD,
+      this.landingURL,
+      directory
+    );
+
+    const newMoments = [...moments];
+    this.shuffledMoments = newMoments;
+
+    this.dispatchEvent(
+      new CustomEvent('momentsShuffled', {
+        detail: { moments: this.shuffledMoments },
+      })
+    );
+  }
+
+  private closeBanner() {
+    this.viewMode = 'closed';
+
+    if (this.hideBannerDays > 0) {
+      (window as any)?.Cookies.set('anniv-banner', 'x', {
+        path: '/',
+        expires: this.hideBannerDays,
+        domain: '.archive.org',
+      });
+    }
+  }
+
+  get closeButton() {
+    return html`<button
+      class="close-banner"
+      title="close banner for the day"
+      @click=${() => this.closeBanner()}
```

Comment:
You don't need the full binding here, you can do this:
```suggestion
      @click=${this.closeBanner}
```